src/docs/components/JSXGenerator.js

```diff
@@ -0,0 +1,45 @@
+function isComponent(value) {
+  return value.$$typeof;
+}
+
+function propToString(prop) {
+  if (isComponent((prop))) {
+    return generateJSX(prop);
+  } else if (Array.isArray(prop)) {
+    return '{[' + prop
+        .map(item => propToString(item))
+        .join(', ') + ']}';
+  } else if (typeof prop === "string") {
+    return JSON.stringify(prop);
+  }
+
+  return '{' + JSON.stringify(prop) + '}';
+}
+
+function generateJSX(component) {
+  if (!isComponent(component)) {
+    return component;
+  }
+
+  const type = component.type.name || component.type;
+  let children = component.props.children || [];
```

Comment:
I'm not sure if you doing it correctly
mapping to array should be done by React.Children.toArray(children)

more:
https://facebook.github.io/react/docs/react-api.html#react.children